### PR TITLE
Add 10th Birthday Blog article

### PR DESCRIPTION
**Pull request by chris-short**
_reviewer's note_: Add a blog article about Kubernetes' 10th birthday.

---

The Docs to Markdown extension generated some errors so I've cleaned those up and have a solid markdown file with no images at all

### CODE REVIEW

File: content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+The community building the project has also expanded immensely over the last decade. Using [DevStats](https://k8s.devstats.cncf.io/d/24/overall-project-statistics?orgId=1), we can see the incredible volume of contribution over the last decade that has made Kubernetes the [second-largest open source project in the world](https://www.cncf.io/reports/kubernetes-project-journey-report/):
+
+* 88,474 contributors
+* 15,121 code committers
+* 4,228,347 contributions
+* 158,530 issues
+* 311,787 pull requests
+
+## Kubernetes Today
```
Comment (sftim) [inline content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md:65]:
```suggestion
## Kubernetes today
```

File: content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+Kubernetes also faces new challenges in serving its vast user base, and the community is adapting accordingly. One example of this is the migration of image hosting to the new, community-owned registry.k8s.io. The egress bandwidth and costs of providing pre-compiled binary images for user consumption have become immense. This new registry change enables the community to continue providing these convenient images in more cost- and performance-efficient ways. Make sure you check out the [blog post](https://kubernetes.io/blog/2022/11/28/registry-k8s-io-faster-cheaper-ga/) and update any automation you have to use registry.k8s.io!
+
+## The Future of Kubernetes
```
Comment (sftim) [inline content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md:73]:
```suggestion
## The future of Kubernetes
```

File: content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+* December 2022 — In 1.26, there was a significant batch and  [job API overhaul](https://kubernetes.io/blog/2022/12/29/scalable-job-tracking-ga/) that paved the way for better support for AI/ML/Batch workloads.
+
+**PS:** Curious to see how far the project has come for yourself? Check out [this tutorial for spinning up a Kubernetes 1.0 cluster](https://github.com/spurin/kubernetes-v1.0-lab) created by community members Carlos Santana, Amim Moises Salum Knabben, and James Spurin.
+
+Kubernetes offers more extension points than we can count. Originally designed to work with Docker and only Docker, now you can plug in any container runtime that adheres to the CRI standard. There are other similar interfaces: CSI for storage and CNI for networking. And that's far from all you can do. In the last decade, whole new patterns have emerged, such as using [Custom Resource Definitions (CRDs)](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) to support third-party controllers - now a huge part of the Kubernetes ecosystem.
```
Comment (sftim) [inline content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md:55]:
```suggestion
Kubernetes offers more extension points than we can count. Originally designed to work with Docker and only Docker, now you can plug in any container runtime that adheres to the CRI standard. There are other similar interfaces: CSI for storage and CNI for networking. And that's far from all you can do. In the last decade, whole new patterns have emerged, such as using [Custom Resource Definitions](/docs/concepts/extend-kubernetes/api-extension/custom-resources/)  (CRDs) to support third-party controllers - now a huge part of the Kubernetes ecosystem.
```

**Review by sftim — DISMISSED** [review]
Mostly mechanical fixes and style nits.

File: content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+* July 2021 — Widely used beta APIs are [removed](/blog/2021/07/14/upcoming-changes-in-kubernetes-1-22/)  in Kubernetes 1.22.
+* May 2022 — Kubernetes 1.24 sees  [beta APIs become disabled by default](/blog/2022/05/03/kubernetes-1-24-release-announcement/) to reduce upgrade conflicts and removal of [Dockershim](/dockershim), leading to [widespread user confusion](https://www.youtube.com/watch?v=a03Hh1kd6KE) (we've since [improved our communication!](https://github.com/kubernetes/community/tree/master/communication/contributor-comms))
+* December 2022 — In 1.26, there was a significant batch and  [Job API overhaul](/blog/2022/12/29/scalable-job-tracking-ga/) that paved the way for better support for AI  /ML / batch workloads.
+
+**PS:** Curious to see how far the project has come for yourself? Check out [this tutorial for spinning up a Kubernetes 1.0 cluster](https://github.com/spurin/kubernetes-v1.0-lab) created by community members Carlos Santana, Amim Moises Salum Knabben, and James Spurin.
```
Comment (sftim) [inline content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md:53]: (nit)
```suggestion
**PS:** Curious to see how far the project has come for yourself? Check out this [tutorial for spinning up a Kubernetes 1.0 cluster](https://github.com/spurin/kubernetes-v1.0-lab) created by community members Carlos Santana, Amim Moises Salum Knabben, and James Spurin.
```

**Comment by chris-short** [comment]
@sftim I put the hold in place as a safety mechanism. Once we're all happy with what's here the hold can disappear. I suspect we'll unhold when we approve

**Comment by sftim** [comment]
/retitle [WIP] Add 10th Birthday Blog article

File: content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+Kubernetes offers more extension points than we can count. Originally designed to work with Docker and only Docker, now you can plug in any container runtime that adheres to the CRI standard. There are other similar interfaces: CSI for storage and CNI for networking. And that's far from all you can do. In the last decade, whole new patterns have emerged, such as using [Custom Resource Definitions (CRDs)](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) to support third-party controllers - now a huge part of the Kubernetes ecosystem.
+
+The community building the project has also expanded immensely over the last decade. Using [DevStats](https://k8s.devstats.cncf.io/d/24/overall-project-statistics?orgId=1), we can see the incredible volume of contribution over the last decade that has made Kubernetes the [second-largest open source project in the world](https://www.cncf.io/reports/kubernetes-project-journey-report/):
+
+* 88,474 contributors
+* 15,121 code committers
+* 4,228,347 contributions
+* 158,530 issues
+* 311,787 pull requests
```
[the remark this answers is not in the document]
Comment (chris-short) [reply content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md]: Done!

File: content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+  * This release also introduced StatefulSets and Pod Disruption Budgets in Beta.
+* April 2017 — [Introduction of Role-Based Access Controls or RBAC](/blog/2017/04/rbac-support-in-kubernetes/).
+* June 2017 — In [Kubernetes 1.7](/blog/2017/06/kubernetes-1-7-security-hardening-stateful-application-extensibility-updates/), Third Party Resources or "TPRs" are replaced with Custom Resource Definitions (CRDs).
+* December 2017 — [Kubernetes 1.9](/blog/2017/12/kubernetes-19-workloads-expanded-ecosystem/) sees the Workloads API becoming GA (Generally Available). The release blog states: _"Deployment and ReplicaSet, two of the most commonly used objects in Kubernetes, are now stabilized after more than a year of real-world use and feedback."_
+* December 2018 — In 1.13, the Container Storage Interface (CSI) reaches GA, kubeadm tool for bootstrapping minimum viable clusters reaches GA, and CoreDNS becomes the default DNS server.
+* September 2019 — [Custom Resource Definitions go GA](/blog/2019/09/18/kubernetes-1-16-release-announcement/)in Kubernetes 1.16.
```
Comment (sftim) [inline content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md:45]: (nit)
```suggestion
* September 2019 — [CustomResourceDefinitions go GA](/blog/2019/09/18/kubernetes-1-16-release-announcement/) in Kubernetes 1.16.
```

File: content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+The project has changed enormously since its original 1.0 release; experiencing a number of big wins such as [Custom Resource Definitions (CRD) going GA in 1.16](/blog/2019/09/18/kubernetes-1-16-release-announcement/) or [full dual stack support launching in 1.23](/blog/2021/12/08/dual-stack-networking-ga/) and community "lessons learned" from the [removal of widely used beta APIs in 1.22](/blog/2021/07/14/upcoming-changes-in-kubernetes-1-22/) or the deprecation of [Dockershim](/blog/2020/12/02/dockershim-faq/).
+
+Some notable updates, milestones and events since 1.0 include:
+
+* December 2016 - [Kubernetes 1.5](/blog/2016/12/kubernetes-1-5-supporting-production-workloads/)introduces runtime pluggability with initial CRI support and alpha Windows node support. OpenAPI also appears for the first time, paving the way for clients to be able to discover extension APIs.
+  * This release also introduced StatefulSets and Pod Disruption Budgets in Beta.
```
Comment (sftim) [inline content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md:40]: (nit)
```suggestion
  * This release also introduced StatefulSets and PodDisruptionBudgets in Beta.
```

File: content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+* July 2021 — Widely used beta APIs are [removed](/blog/2021/07/14/upcoming-changes-in-kubernetes-1-22/)  in Kubernetes 1.22.
+* May 2022 — Kubernetes 1.24 sees  [beta APIs become disabled by default](/blog/2022/05/03/kubernetes-1-24-release-announcement/) to reduce upgrade conflicts and removal of [Dockershim](/dockershim), leading to [widespread user confusion](https://www.youtube.com/watch?v=a03Hh1kd6KE) (we've since [improved our communication!](https://github.com/kubernetes/community/tree/master/communication/contributor-comms))
+* December 2022 — In 1.26, there was a significant batch and  [Job API overhaul](/blog/2022/12/29/scalable-job-tracking-ga/) that paved the way for better support for AI  /ML / batch workloads.
+
+**PS:** Curious to see how far the project has come for yourself? Check out this [tutorial for spinning up a Kubernetes 1.0 cluster](https://github.com/spurin/kubernetes-v1.0-lab) created by community members Carlos Santana, Amim Moises Salum Knabben, and James Spurin.
+
```
Comment (sftim) [inline content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md:54]:
```suggestion

---

```

(renders to `<hr/>`)

It'll make the tutorial stand out IMO.

File: content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+Kubernetes offers more extension points than we can count. Originally designed to work with Docker and only Docker, now you can plug in any container runtime that adheres to the CRI standard. There are other similar interfaces: CSI for storage and CNI for networking. And that's far from all you can do. In the last decade, whole new patterns have emerged, such as using [Custom Resource Definitions](/docs/concepts/extend-kubernetes/api-extension/custom-resources/)  (CRDs) to support third-party controllers - now a huge part of the Kubernetes ecosystem.
```
Comment (sftim) [inline content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md:55]: (nit)
```suggestion
Kubernetes offers more extension points than we can count. Originally designed to work with Docker and only Docker, now you can plug in any container runtime that adheres to the CRI standard. There are other similar interfaces: CSI for storage and CNI for networking. And that's far from all you can do. In the last decade, whole new patterns have emerged, such as using [CustomResourceDefinitions](/docs/concepts/extend-kubernetes/api-extension/custom-resources/)  (CRDs) to support third-party controllers - now a huge part of the Kubernetes ecosystem.
```

File: content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+
+Some notable updates, milestones and events since 1.0 include:
+
+* December 2016 - [Kubernetes 1.5](/blog/2016/12/kubernetes-1-5-supporting-production-workloads/)introduces runtime pluggability with initial CRI support and alpha Windows node support. OpenAPI also appears for the first time, paving the way for clients to be able to discover extension APIs.
+  * This release also introduced StatefulSets and Pod Disruption Budgets in Beta.
+* April 2017 — [Introduction of Role-Based Access Controls or RBAC](/blog/2017/04/rbac-support-in-kubernetes/).
+* June 2017 — In [Kubernetes 1.7](/blog/2017/06/kubernetes-1-7-security-hardening-stateful-application-extensibility-updates/), Third Party Resources or "TPRs" are replaced with Custom Resource Definitions (CRDs).
```
Comment (sftim) [inline content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md:42]:
```suggestion
* June 2017 — In [Kubernetes 1.7](/blog/2017/06/kubernetes-1-7-security-hardening-stateful-application-extensibility-updates/), ThirdPartyResources or "TPRs" are replaced with CustomResourceDefinitions (CRDs).
```

Optionally, ThirdPartyResource could link to https://github.com/kubernetes/design-proposals-archive/blob/main/api-machinery/extending-api.md

**Review by sftim — COMMENTED** [review]
More nit-level feedback. Doesn't block publication.

**Comment by sftim** [comment]
Still WIP? Does this need more work?

You can squash this down to 1 commit @chris-short but I won't in any way insist on it.

**Comment by chris-short** [comment]
@sftim @mrbobbytables I tried squashing it, but, of course, I did something wrong, and now it's gone. I then noticed Tim's additional change requests and I'm stumped...

**Comment by mrbobbytables** [comment]
yay for spelling mistakes on mobile lol
/label tide/merge-method-squash

File: content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md [hunk 11]
```diff
@@ -0,0 +1,206 @@
...
+applications in much more portable and repeatable ways was suddenly possible, but the question of
+scale remained.
+
+Google's Borg system for managing application orchestration at scale had adopted Linux containers as
+they were developed in the mid-2000s. Since then, the company had also started working on a new
+version of the system called "Omega." Engineers at Google who were familiar with the Borg and Omega
+systems saw the popularity of containerization driven by Docker. They recognized not only the need
+for an open source container orchestration system but its "inevitability," as described by Brendan
+Burns in
+[this blog post](https://kubernetes.io/blog/2018/07/20/the-history-of-kubernetes-the-community-behind-it/).
```
Comment (dipesh-rawat) [inline content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md:80]: Consider using site-relative link here.

```suggestion
[this blog post](/blog/2018/07/20/the-history-of-kubernetes-the-community-behind-it/).
```

File: content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md [hunk 12]
```diff
@@ -0,0 +1,206 @@
...
+* June 2017 — In [Kubernetes 1.7](/blog/2017/06/kubernetes-1-7-security-hardening-stateful-application-extensibility-updates/), Third Party Resources or "TPRs" are replaced with Custom Resource Definitions (CRDs).
+* December 2017 — [Kubernetes 1.9](/blog/2017/12/kubernetes-19-workloads-expanded-ecosystem/) sees the Workloads API becoming GA (Generally Available). The release blog states: _"Deployment and ReplicaSet, two of the most commonly used objects in Kubernetes, are now stabilized after more than a year of real-world use and feedback."_
+* December 2018 — In 1.13, the Container Storage Interface (CSI) reaches GA, kubeadm tool for bootstrapping minimum viable clusters reaches GA, and CoreDNS becomes the default DNS server.
+* September 2019 — [Custom Resource Definitions go GA](/blog/2019/09/18/kubernetes-1-16-release-announcement/)in Kubernetes 1.16.
+* August 2020 — [Kubernetes 1.19](/blog/2016/12/kubernetes-1-5-supporting-production-workloads/) increases the support window for releases to 1 year.
+* December 2020 — [Dockershim is deprecated](https://kubernetes.io/blog/2020/12/18/kubernetes-1.20-pod-impersonation-short-lived-volumes-in-csi/)  in 1.20
```
Comment (dipesh-rawat) [inline content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md:123]: Consider using site-relative link here.

```suggestion
* December 2020 — [Dockershim is deprecated](/blog/2020/12/18/kubernetes-1.20-pod-impersonation-short-lived-volumes-in-csi/)  in 1.20
```

File: content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md [hunk 13]
```diff
@@ -0,0 +1,206 @@
...
+## Kubernetes today
+
+<img src="welcome.jpg" alt="KubeCon NA 2023" class="left" style="max-width: 20em; margin: 1em">
+
+Since its early days, the project has seen enormous growth in technical capability, usage, and
+contribution. The project is still actively working to improve and better serve its users.
+
+In the upcoming 1.31 release, the project will celebrate the culmination of an important long-term
+project: the removal of in-tree cloud provider code. In this
+[largest migration in Kubernetes history](https://kubernetes.io/blog/2024/05/20/completing-cloud-provider-migration/),
```
Comment (dipesh-rawat) [inline content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md:159]: Consider using site-relative link here.

```suggestion
[largest migration in Kubernetes history](/blog/2024/05/20/completing-cloud-provider-migration/),
```

File: content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md [hunk 14]
```diff
@@ -0,0 +1,206 @@
...
+base. Vendor-specific capabilities can now be better served by other pluggable extensibility
+features or patterns, such as
+[Custom Resource Definitions (CRDs)](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)
+or API standards like the [Gateway API](https://gateway-api.sigs.k8s.io/).
+Kubernetes also faces new challenges in serving its vast user base, and the community is adapting
+accordingly. One example of this is the migration of image hosting to the new, community-owned
+registry.k8s.io. The egress bandwidth and costs of providing pre-compiled binary images for user
+consumption have become immense. This new registry change enables the community to continue
+providing these convenient images in more cost- and performance-efficient ways. Make sure you check
+out the [blog post](https://kubernetes.io/blog/2022/11/28/registry-k8s-io-faster-cheaper-ga/) and
```
Comment (dipesh-rawat) [inline content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md:173]: Consider using site-relative link here.

```suggestion
out the [blog post](/blog/2022/11/28/registry-k8s-io-faster-cheaper-ga/) and
```

**Comment by mrbobbytables** [comment]
/approve
/lgtm
/hold
I'll leave it to @sftim to release the hold or merge other suggestions beforehand and re-lgtm

File: content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md [hunk 15]
```diff
@@ -0,0 +1,208 @@
...
+
+Google's Borg system for managing application orchestration at scale had adopted Linux containers as
+they were developed in the mid-2000s. Since then, the company had also started working on a new
+version of the system called "Omega." Engineers at Google who were familiar with the Borg and Omega
+systems saw the popularity of containerization driven by Docker. They recognized not only the need
+for an open source container orchestration system but its "inevitability," as described by Brendan
+Burns in this [blog post](/blog/2018/07/20/the-history-of-kubernetes-the-community-behind-it/). That
+realization in the fall of 2013 inspired a small team to start working on a project that would later
+become **Kubernetes**. That team included Joe Beda, Brendan Burns, Craig McLuckie, Ville Aikas, Tim
+Hockin, Dawn Chen, Brian Grant, and Daniel Smith.
```
Comment (mengjiao-liu) [inline content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md:60]: Why is this paragraph the same as L74-83 below?

Comment (mengjiao-liu) [reply content/en/blog/_posts/2024-06-06-10-Years-of-Kubernetes/index.md:60]: PR https://github.com/kubernetes/website/pull/46696 has resolved it.

**Comment by ricardoamaro** [comment]
Seems like despite having made a small contribution in https://docs.google.com/document/d/12Z43mBReJTesjbw2SbbGNPTiqYdwoBxFPC9iLRELGdc I came late to the PR party.  Thank you all for merging it!

**Comment by ricardoamaro** [comment]
Filed a PR for a key milestone: March 2018 Kubernetes becomes the first graduated CNCF project https://github.com/kubernetes/website/pull/46744

**Pull request merged**